### PR TITLE
Add options to delete the word before and after the cursor

### PR DESCRIPTION
**Pull request by learn-more**
The hotkeys ctrl+backspace and ctrl+delete can be used
#99
#106

### CODE REVIEW

File: src/myframe.h [hunk 1]
```diff
@@ -381,6 +381,10 @@ struct MyFrame : wxFrame {
             MyAppend(
                 editmenu, A_BACKSPACE, _(L"Delete Before\tBACK"),
                 _(L"Deletes the column of cells before the selected grid line, or the row above"));
+            MyAppend(editmenu, A_DELETE_WORD, _(L"&Delete After\tCTRL+DEL"),
+                     _(L"Deletes the entire word after the cursor"));
+            MyAppend(editmenu, A_BACKSPACE_WORD, _(L"Delete Before\tCTRL+BACK"),
+                     _(L"Deletes the entire word before the cursor"));
```
Comment (aardappel) [inline src/myframe.h]: Add `Word` to these 2 menu strings to differentiate them from the non-ctrl version?

File: src/text.h [hunk 2]
```diff
@@ -344,6 +344,14 @@ struct Text {
         while (s.cursorend < (int)t.Len() && wxIsalnum(t[s.cursorend])) s.cursorend++;
     }
 
+    void SelectWordBefore(Selection &s) {
+        if (s.cursor <= 1) return;
+        s.cursorend = s.cursor--;
+        if (!wxIsalnum(t[s.cursor])) return;
+        while (s.cursor > 0 && wxIsalnum(t[s.cursor - 1])) s.cursor--;
+        while (s.cursorend < (int)t.Len() && wxIsalnum(t[s.cursorend])) s.cursorend++;
```
Comment (aardappel) [inline src/text.h]: These 3 lines are the same as the previous function.. pull them out into a helper function? or add a `bool before` arg? Whatever you think is nicer.

File: src/text.h [hunk 3]
```diff
@@ -387,6 +395,19 @@ struct Text {
                 --s.cursorend;
             };
     }
+    void DeleteWord(Selection &s) {
+        SelectWord(s);
+        if (!RangeSelRemove(s))
+            if (s.cursor < (int)t.Len()) { t.Remove(s.cursor, 1); };
```
Comment (aardappel) [inline src/text.h]: Call `Delete`? Or call `SelectWord` in the caller, and remove this function entirely?

File: src/text.h [hunk 4]
```diff
@@ -387,6 +395,19 @@ struct Text {
...
+        if (!RangeSelRemove(s))
+            if (s.cursor < (int)t.Len()) { t.Remove(s.cursor, 1); };
+    }
+    void BackspaceWord(Selection &s) {
+        SelectWordBefore(s);
+        if (!RangeSelRemove(s))
+            if (s.cursor > 0) {
+                t.Remove(--s.cursor, 1);
+                --s.cursorend;
+            };
```
Comment (aardappel) [inline src/text.h]: Same here, call `Backspace` and/or move `SelectWordBefore` to caller.

File: src/myframe.h (hunk 1 above)
Comment (learn-more) [reply src/myframe.h]: Totally missed this, thanks

File: src/text.h (hunk 2 above)
Comment (learn-more) [reply src/text.h]: Yeah, I more or less kept to the style of the surrounding code.
The other functions were also roughly the same so I duplicated the code here as well.

Comment (aardappel) [reply src/text.h]: Well, the style of the surrounding code is to not repeat yourself :)

**Comment by learn-more** [comment]
Much cleaner now, thanks for the feedback!

File: src/document.h [hunk 5]
```diff
@@ -1126,6 +1126,26 @@ struct Document {
                 ZoomOutIfNoGrid(dc);
                 return nullptr;
 
+            case A_BACKSPACE_WORD:
+                if (c && selected.TextEdit()) {
```
Comment (aardappel) [inline src/document.h:1130]: these two actions can be moved to below around line 1569 where it says `if (!selected.TextEdit()) return _(L"only works in cell text mode");`, so this `if` can be omitted.

**Comment by aardappel** [comment]
Thanks!

**Pull request merged**